src/input.rs

```diff
@@ -67,6 +67,16 @@ impl InputDescription {
             Some(kind) => format!("{} '{}'", kind.to_lowercase(), self.name),
         })
     }
+
+    pub fn file_num_lines(&self) -> Result<usize> {
+        let file = File::open(&self.name).map_err(|e| format!("file open error : {}", e))?;
+        let file_reader = BufReader::new(file);
+        let mut total_lines = 0;
+        for _ in file_reader.lines() {
+            total_lines += 1;
+        }
+        Ok(total_lines)
+    }
```

Comment:
This will slow down `bat` considerably and it will most probably break some use cases like reading from a (named) pipe. We do not want (and for streaming reasons can not) open the file to count the number of lines before starting to display it.